core/base/mergeTreeClustering/MergeTreeDistance.h

```diff
@@ -0,0 +1,1504 @@
+                       &outputMatching) {
+      Memory m;
+      Timer t_total;
+
+      ftm::FTMTree_MT *tree1 = &(mTree1.tree);
+      ftm::FTMTree_MT *tree2 = &(mTree2.tree);
+
+      // ---------------------
+      // ----- Testing
+      // --------------------
+      testing_ = false;
+      if(testing_) {
+        // progressiveComputation_ = true;
+        // branchDecomposition_ = true;
+        // barycenterMergeTree_ = true;
+        branchDecomposition_ = false;
+        normalizedWasserstein_ = false;
+        keepSubtree_ = true;
+        // parallelize_ = true;
+        epsilonTree1_ = 0;
+        epsilonTree2_ = 0;
+        epsilon2Tree1_ = 0;
+        epsilon2Tree2_ = 0;
+        makeMyTest<dataType>(tree1, tree2);
+        // printTree(tree1); printTree(tree2);
+        // printTreeScalars<dataType>(tree1); printTreeScalars<dataType>(tree2);
+        // tree1->printPairsFromTree<dataType>();
+        // tree2->printPairsFromTree<dataType>();
+
+        // MergeTreeClusteringBarycenter bary;
+        // bary.execute<dataType>(makeMyTest2<dataType>());
+      }
+
+      // ---------------------
+      // ----- Preprocessing
+      // --------------------
+      MergeTree<dataType> tree1Ori = mTree1;
+      MergeTree<dataType> tree2Ori = mTree2;
+      if(saveTree_) {
+        mTree1 = copyMergeTree<dataType>(mTree1);
+        mTree2 = copyMergeTree<dataType>(mTree2);
+        tree1 = &(mTree1.tree);
+        tree2 = &(mTree2.tree);
+      }
+      if(not isCalled_) {
+        verifyMergeTreeStructure<dataType>(tree1);
+        verifyMergeTreeStructure<dataType>(tree2);
+      }
+      if(preprocess_) {
+        treesNodeCorr_ = std::vector<std::vector<int>>(2);
+        preprocessingPipeline<dataType>(mTree1, epsilonTree1_, epsilon2Tree1_,
+                                        epsilon3Tree1_, branchDecomposition_,
+                                        useMinMaxPair_, cleanTree_,
+                                        treesNodeCorr_[0], verbose_);
+        preprocessingPipeline<dataType>(mTree2, epsilonTree2_, epsilon2Tree2_,
+                                        epsilon3Tree2_, branchDecomposition_,
+                                        useMinMaxPair_, cleanTree_,
+                                        treesNodeCorr_[1], verbose_);
+      }
+      tree1 = &(mTree1.tree);
+      tree2 = &(mTree2.tree);
+
+      // ---------------------
+      // ----- Compute Distance
+      // --------------------
+      dataType distance
+        = computeDistance<dataType>(tree1, tree2, outputMatching);
+
+      // ---------------------
+      // ----- Postprocessing
+      // --------------------
+      if(postprocess_) {
+        postprocessingPipeline<dataType>(tree1);
+        postprocessingPipeline<dataType>(tree2);
+        if(branchDecomposition_)
+          convertBranchDecompositionMatching<dataType>(
+            tree1, tree2, outputMatching);
+      }
+
+      if(verbose_ > 0) {
```

Comment:
I believe the internal variable `debugLevel_` is meant to play a very similar (if not identical) role.

Reply:
(it conditions the display of the messages sent with `printMsg()`)

Reply:
Yes it has an identical role. The `verbose` of the mergeTreeClustering filter will be removed and replaced by the use of debugLevel_!